lib/js/sw-base.js

```diff
@@ -0,0 +1,61 @@
+(function (self, localforage) {
+  'use strict';
+
+  self.wpSwManager = {
+    ONE_DAY: 24 * 60 * 60 * 1000, // in ms
+
+    storage: localforage.createInstance({ name: '__wpswmanager'} ),
+
+    checkForUnregistering: function () {
+      if (!self.registration.active) {
+        return Promise.resolve(false);
+      }
+      else {
+        return this.getLastCheck()
+        .then(timestamp => {
+          timestamp = timestamp || 0;
+          var now = Date.now();
+          if (now - timestamp > this.ONE_DAY) {
+            return Promise.all([
+              this.setLastCheck(now),
+              fetch(self.registration.active.scriptURL, { method: 'HEAD' })
+            ])
+            .then(results => {
+              var response = results[1];
+              // If not served as JavaScript, it is the error page from WordPress
+              return response.headers.get('Content-Type').indexOf('javascript') < 0;
```

Comment:
I would `toLowerCase()` before `indexOf`. But I'm not sure how reliable this check will be. If browsers are lenient about enforcing this, then I can imagine it being unreliable.